pkg/apis/eventsource/v1alpha1/types.go

```diff
@@ -863,6 +863,10 @@ type GitlabEventSource struct {
 	// Filter
 	// +optional
 	Filter *EventSourceFilter `json:"filter,omitempty" protobuf:"bytes,12,opt,name=filter"`
+	// List of group IDs or group name like "test".
+	// Group level hook available in Premium and Ultimate Gitlab.
+	// +optional
+	Groups []string `json:"groups,omitempty" protobuf:"bytes,13,rep,name=groups"`
```

Comment:
Not familiar with Gitlab, is the field `Projects` still required when `Groups` field is present? What is the relationship between them?

Reply:
Group hooks and project hooks could exist at the same time. Group Hooks are different from project Hooks that are limited to one project.  [https://docs.gitlab.com/ee/api/groups.html#hooks](url)
From my perspective, both of them (Group & Project) should be optional. However, at least one of them should be set.
Only  `premium` and `ultimate ` version Gitlab supports Group level hook. Since all projects under the group could trigger group hook. I avoid to make it too many api calls (which is truly ugly) by using the premium feature of Gitlab.